pkg/msr/msr_linux.go

```diff
@@ -17,32 +16,27 @@ import (
 // CPUs is a slice of the various cpus to read or write the MSR to.
 type CPUs []uint64
 
-// AllCPUs is a helper variable that lists all the cpus available on the machine.
-var AllCPUs CPUs
-
-func init() {
-	var errs []error
-
-	if AllCPUs, errs = getAllCPUs(); errs != nil {
-		// Should I fatal here? panic?
-		log.Print("Failed to find all CPUs from /dev/cpu")
-		log.Print(errs)
+func AllCPUs() (CPUs, error) {
+	c, errs := GlobCPUs("*")
+	if errs != nil || len(c) == 0 {
+		return nil, fmt.Errorf("error finding all cpus, maybe try modprobe msr? : %v", errs)
 	}
+	return c, nil
 }
 
 // GlobCPUs allow the user to specify CPUs using a glob as one would in /dev/cpu
```

Comment:
Why not just Glob
msr.Glob()

?

Reply:
Glob what? All MSR values possible on my platform? Or all CPUs? Or all whatevers? Not obvious.